src/node_buffer.cc

```diff
@@ -949,6 +983,8 @@ void IndexOfBuffer(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(Environment::GetCurrent(args), args[0]);
```

Comment:
Need to place `THROW_AND_RETURN_UNLESS_BUFFER(Environment::GetCurrent(args), args[1]);`. The value passed to `lastIndexOf()` could have had it's prototype messed with, resulting in the following:
```
$ ./node -e 'Buffer(4).lastIndexOf({__proto__: Buffer.prototype})'
node: ../src/node_buffer.cc:985: void node::Buffer::IndexOfBuffer(const FunctionCallbackInfo<v8::Value> &): Assertion `(args[1])->IsUint8Array()' failed.
Aborted (core dumped)
```
And even though they're messing around in a way they shouldn't, it still shouldn't be possible to cause an abort using the JS API.